packages/mui-base/src/useSwitch/useSwitch.ts

```diff
@@ -115,11 +155,27 @@ export default function useSwitch(props: UseSwitchParameters): UseSwitchReturnVa
     onBlur: createHandleBlur(otherProps),
   });
 
+  const getSwitchRootProps: UseSwitchReturnValue['getSwitchRootProps'] = (otherProps = {}) => ({
```

Comment:
The switch root is a span by default, so props like `type`, `required` or `disabled` are not valid. Instead, use ARIA attributes (such as `aria-disabled`, etc.)

Reply:
added: `ariaReadOnly`, `ariaDisabled`, `ariaRequired`, `ariaType`

removed: `checked`, `defaultChecked`, `disabled`, `readOnly`, `required`, `type`: